tests/cli_terraform_test.go

```diff
@@ -0,0 +1,143 @@
+			t.Fatalf("Failed to change back to the starting directory: %v", err)
+		}
+	}()
+
+	// Define the work directory and change to it
+	workDir := "../examples/quick-start-simple"
+	if err := os.Chdir(workDir); err != nil {
+		t.Fatalf("Failed to change directory to %q: %v", workDir, err)
+	}
+
+	// Find the binary path for "atmos"
+	binaryPath, err := exec.LookPath("atmos")
+	if err != nil {
+		t.Fatalf("Binary not found: %s. Current PATH: %s", "atmos", pathManager.GetPath())
+	}
+	// Run terraform apply for prod environment
+	runTerraformApply(t, binaryPath, "prod")
+	verifyStateFilesExist(t, []string{"./components/terraform/weather/terraform.tfstate.d/prod-station"})
+	runCLITerraformCleanComponent(t, binaryPath, "prod")
+	verifyStateFilesDeleted(t, []string{"./components/terraform/weather/terraform.tfstate.d/prod-station"})
+
+	// Run terraform apply for dev environment
+	runTerraformApply(t, binaryPath, "dev")
+
+	// Run terraform apply for prod environment
+	runTerraformApply(t, binaryPath, "prod")
+
+	// Verify if state files exist before cleaning
+	stateFiles := []string{
+		"./components/terraform/weather/.terraform",
+		"./components/terraform/weather/terraform.tfstate.d",
+		"./components/terraform/weather/.terraform.lock.hcl",
+	}
+	verifyStateFilesExist(t, stateFiles)
+
+	// Run terraform clean
+	runTerraformClean(t, binaryPath)
+
+	// Verify if state files have been deleted after clean
+	verifyStateFilesDeleted(t, stateFiles)
+
+}
+
+// runTerraformApply runs the terraform apply command for a given environment.
+func runTerraformApply(t *testing.T, binaryPath, environment string) {
+	cmd := exec.Command(binaryPath, "terraform", "apply", "station", "-s", environment)
+	envVars := os.Environ()
+	envVars = append(envVars, "ATMOS_COMPONENTS_TERRAFORM_APPLY_AUTO_APPROVE=true")
+	cmd.Env = envVars
+
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	t.Log(stdout.String())
+	if err != nil {
+		t.Fatalf("Failed to run terraform apply station -s %s: %v", environment, stderr.String())
+	}
+}
+
+// verifyStateFilesExist checks if the state files exist before cleaning.
+func verifyStateFilesExist(t *testing.T, stateFiles []string) {
+	for _, file := range stateFiles {
+		fileAbs, err := filepath.Abs(file)
+		if err != nil {
+			t.Fatalf("Failed to resolve absolute path for %q: %v", file, err)
+		}
+		if _, err := os.Stat(fileAbs); errors.Is(err, os.ErrNotExist) {
+			t.Errorf("Expected file to exist before cleaning: %q", fileAbs)
+		}
+	}
+}
+
+// runTerraformClean runs the terraform clean command.
+func runTerraformClean(t *testing.T, binaryPath string) {
+	cmd := exec.Command(binaryPath, "terraform", "clean", "--force")
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	t.Logf("Clean command output:\n%s", stdout.String())
+	if err != nil {
+		t.Fatalf("Failed to run terraform clean: %v", stderr.String())
+	}
+}
```

Comment:
Why are we implementing go functions here, instead of updating test-cases YAML? ...unless the goal is to test the tests, these should be in YAML.

Reply:
As you explained to me via Slack, this is testing a sequence of commands, which is not currently supported by the test-cases YAML.